geo/src/algorithm/lines_iter.rs

```diff
@@ -0,0 +1,307 @@
+use crate::{
+    CoordNum, Coordinate, Line, LineString, MultiLineString, MultiPolygon, Polygon, Rect, Triangle,
+};
+use core::slice;
+use std::fmt::Debug;
+use std::iter;
+
+/// Iterate over lines of a geometry.
+pub trait LinesIter<'a> {
```

Comment:
What do you think about implementing `LinesIter` for `Point` and `MultiPoint` that return empty iterators? Then we could implement `LinesIter` for `GeometryCollection`. I don't feel strongly about it, just an idea.

Reply:
It **would** be nice to have LinesIter on GeometryCollection, but I can't decide how surprising it would be for Point/MultiPoint...

My instinct is to hold off on including it for now. It would be a strictly additive change at some point if we wanted to add it.

Reply:
I was also conflicted about `Point` and `MultiPoint`. There is the better ergonomics of being able to iterate over the lines of a `Geometry`, but clients lose some help from compiler in catching cases where they probably don't want to iterate over lines in a `Point`.

I'd vote to not add it right now.

Reply:
Sounds great, thanks for the feedback!